internal/services/compute/snapshot_resource.go

```diff
@@ -179,73 +181,74 @@ func resourceSnapshotRead(d *pluginsdk.ResourceData, meta interface{}) error {
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.SnapshotID(d.Id())
+	id, err := snapshots.ParseSnapshotID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	resp, err := client.Get(ctx, *id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			log.Printf("[INFO] Error reading Snapshot %q - removing from state", d.Id())
 			d.SetId("")
 			return nil
 		}
 
-		return fmt.Errorf("making Read request on Snapshot %q: %+v", id.Name, err)
+		return fmt.Errorf("making Read request on Snapshot %q: %+v", id.SnapshotName, err)
 	}
 
-	d.Set("name", id.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
-	if location := resp.Location; location != nil {
-		d.Set("location", azure.NormalizeLocation(*location))
-	}
+	d.Set("name", id.SnapshotName)
+	d.Set("resource_group_name", id.ResourceGroupName)
 
-	if props := resp.SnapshotProperties; props != nil {
-		if data := props.CreationData; data != nil {
+	if model := resp.Model; model != nil {
+		if location := model.Location; location != "" {
+			d.Set("location", azure.NormalizeLocation(location))
+		}
+
+		if props := model.Properties; props != nil {
+			data := props.CreationData
 			d.Set("create_option", string(data.CreateOption))
 
-			if accountId := data.StorageAccountID; accountId != nil {
+			if accountId := data.StorageAccountId; accountId != nil {
 				d.Set("storage_account_id", accountId)
 			}
-		}
 
-		if props.DiskSizeGB != nil {
-			d.Set("disk_size_gb", int(*props.DiskSizeGB))
-		}
+			if props.DiskSizeGB != nil {
+				d.Set("disk_size_gb", int(*props.DiskSizeGB))
+			}
 
-		if err := d.Set("encryption_settings", flattenSnapshotDiskEncryptionSettings(props.EncryptionSettingsCollection)); err != nil {
-			return fmt.Errorf("setting `encryption_settings`: %+v", err)
-		}
+			if err := d.Set("encryption_settings", flattenSnapshotDiskEncryptionSettings(props.EncryptionSettingsCollection)); err != nil {
+				return fmt.Errorf("setting `encryption_settings`: %+v", err)
+			}
 
-		trustedLaunchEnabled := false
-		if securityProfile := props.SecurityProfile; securityProfile != nil {
-			if securityProfile.SecurityType == compute.DiskSecurityTypesTrustedLaunch {
-				trustedLaunchEnabled = true
+			trustedLaunchEnabled := false
+			if securityProfile := props.SecurityProfile; securityProfile != nil && securityProfile.SecurityType != nil {
+				if *securityProfile.SecurityType == snapshots.DiskSecurityTypesTrustedLaunch {
+					trustedLaunchEnabled = true
+				}
```

Comment:
(as above) this can be reduced to:

```suggestion
				trustedLaunchEnabled = *securityProfile.SecurityType == snapshots.DiskSecurityTypesTrustedLaunch
```